projects/storefrontlib/src/lib/store-finder/services/store-finder.service.spec.ts

```diff
@@ -24,11 +25,11 @@ describe('StoreFinderService', () => {
     nativeWindow: {
       navigator: {
         geolocation: {
-          getCurrentPosition: callback => {
-            callback({
-              coords: longitudeLatitude
-            });
-          }
+          watchPosition: callback => {
+            callback({ coords: longitudeLatitude });
+            return geolocationWatchId;
+          },
+          clearWatch: () => {}
```

Comment:
Make the expectation on invoking clearWatch() in your tests when whachId is defined.
Also, since watchId is a private variable the proper testcase would not be:

service.watchId = 1;
service.findStores();
expect(clearWatch).toHaveBeenCalled();

but rather:

service.findeStore(..., true);// find stores with use of myLocation which updates watchIId
service.findStores();
expect(clearWatch).toHaveBeenCalled();